cvmfs/sync_item.cc

```diff
@@ -216,6 +217,18 @@ std::string SyncItem::GetScratchPath() const {
   return union_engine_->scratch_path() + relative_path;
 }
 
+void SyncItem::CheckMarkerFiles() {
+  if (IsRegularFile()) {
+    CheckGraft();
+  } else if (IsDirectory()) {
+    CheckCatalogMarker();
+  }
+}
+
+void SyncItem::CheckCatalogMarker() {
+  has_catalog_marker_ = FileExists(GetUnionPath() + "/.cvmfscatalog");
```

Comment:
`.cvmfsautocatalog` is not supposed to trigger nested catalog creation. It is dropped by the `CatalogBalancer` during `cvmfs_server publish`. I think should leave it out of those mechanics.